### PR TITLE
#11 Submit cyclone bom to IQ

### PR DESCRIPTION
**Pull request by klehrke-sonatype**

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -212,6 +213,8 @@ github.com/smartystreets/goconvey v1.6.4/go.mod h1:syvi0/a8iFYH4r/RixwvyeAJjdLS9
 github.com/soheilhy/cmux v0.1.4/go.mod h1:IM3LyeVVIOuxMH7sFAkER9+bJ4dT7Ms6E4xg4kGIyLM=
 github.com/sonatype-nexus-community/go-sona-types v0.0.8 h1:6xb9BIC2w3y4kF/xQA25Zs6Yrl8ZqFkfH77AKPaG4RA=
 github.com/sonatype-nexus-community/go-sona-types v0.0.8/go.mod h1:uou8FGf9R5Nz1c6BfSM3v9K7g0R6faTYoxLh9Ybeht8=
```
Comment (bhamail) [inline go.sum:215]: minor: I think `go mod tidy` would clean (remove) the old hashes for `g-s-t` if they're no longer used.

**Review by bhamail — COMMENTED** [review]
Nice work! minor comment re: `go mod tidy`

File: go.sum (hunk 1 above)
Comment (klehrke-sonatype) [reply go.sum:215]: i've run go mod tidy, it doesn't make any changes to the go.mod / go.sum file. :(

File: main.go [hunk 2]
```diff
@@ -93,3 +88,44 @@ func main() {
 
 	os.Exit(0)
 }
+
+func generateCycloneDx(config config.Config, lResults *linker.Results) {
```
Comment (DarthHater) [inline main.go:92]: Personally I would name this `auditWithIQ`, the SBOM is just kinda the way in which you do.

File: main.go [hunk 3]
```diff
@@ -93,3 +88,44 @@ func main() {
...
+		dx := cyclonedx.Default(logger.GetLogger())
+		sbom := dx.FromCoordinates(lResults.Coordinates)
+
+		iqOptions := iq.Options{
+			User:        config.IQConfig.Username,
+			Token:       config.IQConfig.Token,
+			Application: "cheque",
+			Server:      config.IQConfig.Server,
+			Stage:       "build",
+			MaxRetries:  60,
```
Comment (DarthHater) [inline main.go:103]: Good candidate for config

File: main.go [hunk 4]
```diff
@@ -93,3 +88,44 @@ func main() {
...
+
+func generateCycloneDx(config config.Config, lResults *linker.Results) {
+	if config.ChequeConfig.UseIQ {
+		dx := cyclonedx.Default(logger.GetLogger())
+		sbom := dx.FromCoordinates(lResults.Coordinates)
+
+		iqOptions := iq.Options{
+			User:        config.IQConfig.Username,
+			Token:       config.IQConfig.Token,
+			Application: "cheque",
```
Comment (DarthHater) [inline main.go:100]: This is an interesting one, also a good candidate for config. Right now you'd be hard coded to one application in IQ Server.

File: main.go [hunk 5]
```diff
@@ -93,3 +88,44 @@ func main() {
...
+			Server:      config.IQConfig.Server,
+			Stage:       "build",
```
Comment (DarthHater) [inline main.go:102]: Good candidate for config.

File: main.go (hunk 4 above)
Comment (klehrke-sonatype) [reply main.go:100]: that's coming in the next issue/PR that's already made heh

File: go.sum (hunk 1 above)
Comment (DarthHater) [reply go.sum:215]: It has some odd behavior, but `go mod tidy` is a ledger, more than a lock file. Basically just has the sha1s around to validate you got the right package. Not sure why it's keeping old stuff, but that was the problem with a `go.sum` file anyways and why we moved away from scanning it!

**Review by DarthHater — APPROVED** [review]
My head's a bit woozy today, but I'll give it the ole stamp!

**Pull request merged**